CPP-951 Add TypeScript plugin (#338)
Financial-Times/dotcom-tool-kit: plugins/typescript/readme.md

# @dotcom-tool-kit/typescript

Tool Kit plugin to build [TypeScript](https://www.typescriptlang.org) code

## Installation & Usage

With Tool Kit [already set up](https://github.com/financial-times/dotcom-tool-kit#installing-and-using-tool-kit), install this plugin as a dev dependency:

```sh
npm install --save-dev @dotcom-tool-kit/typescript
```

And add it to your repo's `.toolkitrc.yml`:

```yml
plugins:
  - '@dotcom-tool-kit/typescript'
```

## Options

| Key | Description | Default value |
|-|-|-|
| `configPath` | Path to the [TypeScript config file](https://www.typescriptlang.org/tsconfig) | use TypeScript's own [tsconfig.json resolution](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html#using-tsconfigjson-or-jsconfigjson) |

> | `extraArgs` | Extra arguments to pass to the tsc CLI (though prefer configuring via `tsconfig.json` if possible) | `[]`

Author: some [options](https://www.typescriptlang.org/docs/handbook/compiler-options.html) can only be passed as command line arguments (like `--verbose` and `--build`) and can't be set in the `tsconfig.json`. not sure if I need to mention that in the docs seeing as it should be clear to the user when this applies?

> | `extraArgs` | Extra arguments to pass to the tsc CLI (though prefer configuring via `tsconfig.json` if possible) | `[]`

Reviewer: We should change the description to reflect this.

```suggestion
| `extraArgs` | Extra [arguments](https://www.typescriptlang.org/docs/handbook/compiler-options.html) to pass to the tsc CLI that can't be set in `tsconfig.json` | `[]`
```

What do you think?
